File: PDOO/Napakalaki/src/NapakalakiGame/DeathBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author nestor
 */
public class DeathBadConsequence extends NumericBadConsequence {

    private boolean death;

    public DeathBadConsequence(String t, boolean death) {
        super(t, Player.MAXLEVEL, MAXTREASURES, MAXTREASURES);
        this.death = death;
    }

    public boolean getDeath() {
        return death;
    }
    
    public String toString() {
        return "Text = " + super.getText() + ": Estás muerto.";
    }
}

File: PDOO/Napakalaki-Ruby/lib/death_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "bad_consequence"
require_relative "player"

module NapakalakiGame
  
  class DeathBadConsequence < NumericBadConsequence
    
    def initialize (aText)
      super(aText, Player.MaxLevel, @@MAXTREASURES, @@MAXTREASURES)
      @death = true
    end
    
    def getDeath
      @death
    end
    
    def to_s
      return "#{super} Death: true "
    end

  end

end
